File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/interactions/Marche.java
package io.github.alexiscomete.lapinousecond.worlds.buildings.interactions;

import io.github.alexiscomete.lapinousecond.entity.Player;
import io.github.alexiscomete.lapinousecond.worlds.buildings.Building;
import io.github.alexiscomete.lapinousecond.worlds.buildings.BuildingInteraction;
import org.javacord.api.entity.message.MessageBuilder;
import org.javacord.api.entity.message.embed.EmbedBuilder;

public class Marche extends BuildingInteraction {
    public Marche(Building building) {
        super(building);
    }

    @Override
    public void interpret(String[] args) {

    }

    @Override
    public String getHelp() {
        return null;
    }

    @Override
    public String getUsage() {
        return null;
    }

    @Override
    public EmbedBuilder getInfos(Player p) {
        return null;
    }

    @Override
    public MessageBuilder getCompleteInfos(Player p) {
        return null;
    }

    @Override
    public void configBuilding() {

    }
}

File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/interactions/Marche.kt
package io.github.alexiscomete.lapinousecond.worlds.buildings.interactions

import io.github.alexiscomete.lapinousecond.view.ui.longuis.LongCustomUI
import io.github.alexiscomete.lapinousecond.view.ui.longuis.MenuBuilderUI
import io.github.alexiscomete.lapinousecond.view.ui.playerui.PlayerUI
import io.github.alexiscomete.lapinousecond.worlds.buildings.Building
import io.github.alexiscomete.lapinousecond.worlds.buildings.BuildingInteraction

class Marche(building: Building?) : BuildingInteraction(building!!) {
    override fun interpret(args: Array<String>) {}
    override val help: String?
        get() = null
    override val usage: String?
        get() = null

    override fun descriptionShort(): String {
        return "Marché"
    }

    override fun title(): String {
        return "Marché"
    }

    override fun getCompleteInfos(ui: PlayerUI): LongCustomUI {
        return MenuBuilderUI("Marché", "Bienvenue dans le marché !", ui)
    }

    override fun configBuilding() {}
}